components/camel-jaxb/src/main/java/org/apache/camel/converter/jaxb/JaxbDataFormat.java

```diff
@@ -136,6 +136,10 @@ public void marshal(Exchange exchange, Object graph, OutputStream stream) throws
             String charset = exchange.getProperty(Exchange.CHARSET_NAME, String.class);
             if (charset == null) {
                 charset = encoding;
+                //Propagate the encoding of the exchange
```

Comment:
This looks a bit ugly.

Reply:
Maybe we can find a better way.

Reply:
Ok, what is the appropriate way to communicate the encoding?

This is what is done by the file endpoint, and afaik how that is able to accomplish the same feature:
`/**
     * Set up the exchange properties with the options of the file endpoint
     */
    public void configureExchange(Exchange exchange) {
        // Now we just set the charset property here
        if (getCharset() != null) {
            exchange.setProperty(Exchange.CHARSET_NAME, getCharset());
        }
    }`